integration/helper/assertions.go

```diff
@@ -0,0 +1,34 @@
+package helper
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+// Subset returns true if the first array is completely
+// contained in the second array. There must be at least
+// the same number of duplicate values in second as there
+// are in first.
+func Subset[V, W any, Z comparable](t *testing.T, first []V, second []W, firstLookup func(V) Z, secondLookup func(W) Z) bool {
+	t.Helper()
+
+	set := make(map[Z]int)
+	for _, value := range second {
+		set[secondLookup(value)] += 1
+	}
+
+	for _, value := range first {
+		parsedValue := firstLookup(value)
+		if count, found := set[parsedValue]; !found {
+			return false
+		} else if count < 1 {
+			// Provide a useful failure message
+			assert.Contains(t, set, parsedValue)
+			return false
+		} else {
+			set[parsedValue] = count - 1
+		}
+	}
```

Comment:
If `parsedValue` is in `set`, but `count < 1`, then I think `assert.Contains` will pass. Also, this can be simplified. Note that reading a key that doesn't exist in a map is perfectly valid and will return the type's zero value.
```suggestion
		if set[parsedValue] < 1 {
			return false
		} 
		set[parsedValue]--
```

Reply:
Great point, moved the assertion outside of the function. TY